trader/trader.go

```diff
@@ -120,9 +120,11 @@ func (t *Trader) Start() {
 	for {
 		currentUpdateTime := time.Now()
 		if lastUpdateTime.IsZero() || t.timeController.ShouldUpdate(lastUpdateTime, currentUpdateTime) {
+			startUpdate := time.Now()
 			success := t.update()
+			millisForUpdate := time.Since(startUpdate).Milliseconds()
```

Comment:
should reuse `currentUpdateTime` instead of creating a new `startUpdate` time. Idea is to use one synchronized startTime variable wherever possible.

Reply:
Got it. What's the reason behind storing both `millisForUpdate` and `commonProps.SecondsSinceStart`? There's basically just two levels of functional indirection that differentiate these timestamps: triggering the goroutine and calling `SendUpdateEvent`. Is there significant overhead there?

Reply:
millisForUpdate is the time taken for the bot compute the action it wants to perform next, i.e. running the `update()` function (excluding sleep times or time waiting for transaction threads to finish). This can give us very useful performance metrics. For example, v1.11.0 is 10% faster than v1.10.0, or systems running windows is 20% slower than linux, as hypothetical examples.

secondsSinceStart and the different with `currentUpdateTime` tells us how many seconds it has been since the bot first started. This is useful in knowing things like long term stability of the bot. For example, say the bot always crashes if it is run for more than 5 hours -- this is what we can gauge from this metric.

Reply:
Got it, that makes sense -- thanks!